daemon/containerd/image_builder.go

```diff
@@ -41,23 +41,17 @@ import (
 // reference or ID. Every call to GetImageAndReleasableLayer MUST call
 // releasableLayer.Release() to prevent leaking of layers.
 func (i *ImageService) GetImageAndReleasableLayer(ctx context.Context, refOrID string, opts backend.GetImageAndLayerOptions) (builder.Image, builder.ROLayer, error) {
-	if refOrID == "" { // from SCRATCH
-		imgOS := runtime.GOOS
+	if refOrID == "" { // FROM scratch
 		if runtime.GOOS == "windows" {
-			imgOS = "linux"
-		}
-		if opts.Platform != nil {
-			imgOS = opts.Platform.OS
+			// FROM scratch is not currently supported on Windows.
+			return nil, nil, system.ErrNotSupportedOperatingSystem
 		}
-		if !system.IsOSSupported(imgOS) {
+		if opts.Platform != nil && !system.IsOSSupported(opts.Platform.OS) {
```

Comment:
I'm really wondering as well if we need the check here at all; assuming follow-up steps would "inherit" the base image ("scratch") platform, we could make it "if base-image has no platform, and no option is passed, then ..."